File: backend/src/attendance-events/dto/batch-events.dto.ts
import { IsString, IsNotEmpty, IsNumber, IsOptional, IsDateString, IsArray, ValidateNested } from 'class-validator';
import { Type } from 'class-transformer';

export class AttendanceEventItemDto {
    @IsString()
    @IsNotEmpty()
    studentId: string;

    @IsDateString()
    @IsNotEmpty()
    occurredAt: string;

    @IsOptional()
    @IsNumber()
    confidence?: number;

    @IsString()
    @IsNotEmpty()
    idempotencyKey: string;

    @IsOptional()
    @IsString()
    source?: 'edge' | 'manual' | 'import';
}

export class BatchEventsDto {
    @IsString()
    @IsNotEmpty()
    sessionId: string;

    @IsArray()
    @ValidateNested({ each: true })
    @Type(() => AttendanceEventItemDto)
    events: AttendanceEventItemDto[];
}

File: android/app/src/main/kotlin/com/smartpresence/idukay/data/remote/dto/BatchEventsDto.kt
package com.smartpresence.idukay.data.remote.dto

data class BatchEventsRequest(
    val sessionId: String,
    val events: List<AttendanceEventDto>
)

data class AttendanceEventDto(
    val studentId: String,
    val occurredAt: String, // ISO 8601
    val confidence: Float?,
    val idempotencyKey: String,
    val source: String = "edge"
)

data class BatchEventsResponse(
    val inserted: Int,
    val ignored: Int,
    val total: Int
)
